src/main/java/org/aksw/word2vecrestful/word2vec/Word2VecModel.java

```diff
@@ -11,9 +17,79 @@
 
   public Map<String, float[]> word2vec;
   public int vectorSize;
-
+  public TreeSet<VectorDimension> sortedVecDimns;
+  public int[] dimRngIndxDesc;
   public Word2VecModel(final Map<String, float[]> word2vec, final int vectorSize) {
     this.word2vec = word2vec;
     this.vectorSize = vectorSize;
+    this.sortedVecDimns = new TreeSet<>();
+    //fetch vector dimension details
+    fetchVectorDimensions();
+  }
+  /**
+   * Method to fetch and store range of each dimension in the model
+   */
+  private void fetchVectorDimensions() {
+	  float[] dimVals;
+	  VectorDimension vectorDimension;
+	  //for each dimension
+	  for(int i=0;i<vectorSize;i++) {
+		  dimVals = new float[word2vec.size()];
```

Comment:
You could optimize this by reusing the `dimVals` vector. You are creating a new array for every dimension and throw away the old array. That is not really necessary ;)

Reply:
Good point :) It will be taken care of